conftest.py

```diff
@@ -170,3 +149,46 @@ def s3_cluster_dump_url(s3, s3_scratch):
     dump_url = f"{s3_scratch}/cluster_dumps"
     s3.mkdirs(dump_url, exist_ok=True)
     return dump_url
+
+
+# this code was taken from pytest docs
+# https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+
+    # set a report attribute for each phase of a call, which can
+    # be "setup", "call", "teardown"
+
+    setattr(item, "rep_" + rep.when, rep)
+
+
+@pytest.fixture
+def upload_cluster_dump(request, s3_cluster_dump_url, s3_storage_options):
+    @contextlib.contextmanager
+    def _upload_cluster_dump(client, cluster):
+        failed = False
+        # the code below is a workaround to make cluster dumps work with clients in fixtures
+        # and outside fixtures.
+        try:
+            yield
+        except:  # noqa: E722
+            failed = True
+            raise
+        finally:
+            try:
+                # we need this for tests that are not using the client fixture
+                # for those cases request.node.rep_call.failed can't be access.
+                failed_req = request.node.rep_call.failed
```

Comment:
Can this just be
```suggestion
                failed = request.node.rep_call.failed
```
?

Do we need the extra flag?